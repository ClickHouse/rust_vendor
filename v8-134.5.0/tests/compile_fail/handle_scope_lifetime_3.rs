// Copyright 2019-2020 the Deno authors. All rights reserved. MIT license.
pub fn main() {
  let mut isolate = v8::Isolate::new(mock());
  let mut scope1 = v8::HandleScope::new(&mut isolate);

  let _local = {
    let mut _scope2 = v8::EscapableHandleScope::new(&mut scope1);
    v8::Integer::new(&mut scope1, 123)
  };
}

fn mock<T>() -> T {
  unimplemented!()
}
