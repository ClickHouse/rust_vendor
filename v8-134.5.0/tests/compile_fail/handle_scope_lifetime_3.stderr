error[E0499]: cannot borrow `scope1` as mutable more than once at a time
 --> $DIR/handle_scope_lifetime_3.rs:8:22
  |
7 |     let mut _scope2 = v8::EscapableHandleScope::new(&mut scope1);
  |                                                     ----------- first mutable borrow occurs here
8 |     v8::Integer::new(&mut scope1, 123)
  |                      ^^^^^^^^^^^ second mutable borrow occurs here
9 |   };
  |   - first borrow might be used here, when `_scope2` is dropped and runs the `Drop` code for type `EscapableHandleScope`
